crates/zk-por-core/src/bin/gen_mst.rs

```diff
@@ -0,0 +1,155 @@
+use plonky2::{hash::hash_types::HashOut, util::log2_strict};
+use rayon::prelude::*;
+
+use std::{
+    fs,
+    ops::Div,
+    path::{Path, PathBuf},
+    sync::RwLock,
+};
+use tracing::{debug, info, warn};
+use zk_por_core::{
+    account::Account,
+    config::ProverConfig,
+    merkle_sum_tree::MerkleSumTree,
+    parser::read_json_into_accounts_vec,
+    util::{get_node_level, get_recursive_hash_nums},
+    GlobalConfig, GLOBAL_CONFIG, GLOBAL_MST,
+};
+use zk_por_tracing::{init_tracing, TraceConfig};
+
+fn list_json_files(dir: &Path) -> std::io::Result<Vec<PathBuf>> {
+    let mut json_files = Vec::new();
+    if dir.is_dir() {
+        for entry in fs::read_dir(dir)? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_file() {
+                if let Some(extension) = path.extension() {
+                    if extension == "json" {
+                        json_files.push(path);
+                    }
+                }
+            } else if path.is_dir() {
+                json_files.extend(list_json_files(&path)?);
+            }
+        }
+    }
+    json_files.sort();
+    Ok(json_files)
+}
+
```

Comment:
You can move this to the parser and add a multi file parsing option where given a root directory, it returns all the accounts as a vec of accounts

Reply:
We can keep parsing and tree building separate as part of the separation of concerns principle.